strax/storage/files.py

```diff
@@ -364,3 +365,110 @@ def _close(self):
+        strax.DataType.
+    :param dest_directory: Head of a folder whereto write new data. If
+        nothing is specified, write to a temporary directory.
+    :param replace: Delete the source_directory and replace it with it's
+        rechunked version
+    :param compressor: Compressor to be used in saving the rechunked
+        data
+    :param target_size_mb: Target size of chunks (uncompressed). As long
+        as a chunk is smaller than this many MB, keep adding new MBs
+        until the chunk is at least target_size_mb or we run out of
+        chunks.
+    :param rechunk: Do we want to rechunk?
+    :return: Dictionary with some information on the write/load times
+        involved.
+    """
+    if not os.path.exists(source_directory):
+        raise FileNotFoundError(f'No file at {source_directory}')
+    if not replace and dest_directory is None:
+        raise ValueError(f'Specify a destination path <dest_file> when '
+                         f'not replacing the original path')
+    backend_key = os.path.basename(os.path.normpath(source_directory))
+
+    if dest_directory is None and replace:
+        _temp_dir = tempfile.TemporaryDirectory()
+        dest_directory = _temp_dir.name
+    else:
+        _temp_dir = False
+
+
+    if os.path.basename(os.path.normpath(dest_directory)) != backend_key:
+        # New key should be correct! If there is not an exact match,
+        # we want to make sure that we append the backend_key correctly
+        dest_directory = os.path.join(dest_directory, backend_key)
+    backend = strax.FileSytemBackend()
+    meta_data = backend.get_metadata(source_directory)
+    source_compressor = meta_data['compressor']
+
+    if compressor is not None:
+        meta_data['compressor'] = compressor
+    if target_size_mb is not None:
+        meta_data['chunk_target_size_mb'] = target_size_mb
+
+    data_loader = backend.loader(source_directory)
+
+    load_time_seconds = []
+    def loader():
+        """Wrapped loader for bookkeeping load time"""
+        while True:
+            try:
+                t0 = time.time()
+                data = next(data_loader)
+                # Update target chunk size for re-chunking
+                data.target_size_mb = meta_data['chunk_target_size_mb']
+                load_time_seconds.append(time.time()-t0)
+            except StopIteration:
+                return
+            yield data
+    print(f'Rechunking {source_directory} to {dest_directory}')
+    saver = backend._saver(dest_directory, metadata=meta_data)
+
+    write_time_start = time.time()
+    saver.save_from(loader(), rechunk=rechunk)
+    load_time = sum(load_time_seconds)
+    write_time = time.time() - write_time_start - load_time
+
+    if replace:
+        print(f'move {dest_directory} to {source_directory}')
+        shutil.rmtree(source_directory)
+        shutil.move(dest_directory, source_directory)
+    if _temp_dir:
+        _temp_dir.cleanup()
```

Comment:
So this means if replace=False and dest_directory=None we are doing a try run in which case the re-compressed data is not stored at all?

Reply:
I guess I should just throw an error if we add such a combination of arguments, it would now just fail on this line:
https://github.com/AxFoundation/strax/pull/686/files#diff-774d8745c27f52bd7b040501d235f6a641688e5b339e7bf844e09ff0f62235c4R425

Reply:
oh I already raised a warning for this